extensions/wikia/CuratedContent/js/CuratedContentManagmentTool.js

```diff
@@ -2,416 +2,414 @@
+				articleNotFoundError = msg('wikiacuratedcontent-content-articlenotfound-error'),
+				emptyLabelError = msg('wikiacuratedcontent-content-emptylabel-error'),
+				tooLongLabelError = msg('wikiacuratedcontent-content-toolonglabel-error'),
+				videoNotSupportedError = msg('wikiacuratedcontent-content-videonotsupported-error'),
+				notSupportedType = msg('wikiacuratedcontent-content-notsupportedtype-error'),
+				noCategoryInTag = msg('wikiacuratedcontent-content-nocategoryintag-error'),
+				imageMissingError = msg('wikiacuratedcontent-content-imagemissing-error'),
+				addItem = d.getElementById('addItem'),
+				addSection = d.getElementById('addSection'),
+				$save = $(d.getElementById('save')),
+				form = d.getElementById('contentManagmentForm'),
+				$form = $(form),
+				ul = form.getElementsByTagName('ul')[0],
+				$ul = $(ul),
+				maxAllowedLength = 48, // it's derived from MAX_LABEL_LENGTH in CuratedContentSpecialController
+
+				setup = function (elem) {
+					(elem || $ul.find('.item-input')).autocomplete({
+						serviceUrl: wgServer + wgScript,
+						params: {
+							action: 'ajax',
+							rs: 'getLinkSuggest',
+							format: 'json'
+							//ns: categoryId
+						},
+						appendTo: form,
+						onSelect: function () {
+							$ul.find('input:focus').next().focus();
+						},
+						fnPreprocessResults: function (data) {
+							return data;
+						},
+						deferRequestBy: 50,
+						minLength: 3,
+						skipBadQueries: true // BugId:4625 - always send the request even if previous one returned no suggestions
+					});
+					// validate form on init
+					checkForm();
+				},
+				addNew = function (row, elem) {
+					var cat;
+
+					if (elem) {
+						elem.after(row);
+						cat = elem.next().find('.item-input');
+					} else {
+						$ul.append(row);
+						cat = $ul.find('.item-input:last');
 					}
-					// check if field value is too long
-					if (optionLimit && val.length > maxAllowedLength) {
-						$this
-							.addClass('error')
-							.popover('destroy')
-							.popover({
-								content: tooLongLabelError
-							});
-						return true;
+
+					setup(cat);
+					cat.focus();
+
+					$ul.sortable('refresh');
+				},
+
+				/**
+				 * Validate input elements
+				 *
+				 * @param elements
+				 * @param options array consists of ['checkEmpty', 'required', 'limit']
+				 */
+				checkInputs = function (elements, options) {
+					var cachedVals = [],
+						optionCheckEmpty, optionRequired, optionLimit;
+
+					if (Array.isArray(options)) {
+						optionCheckEmpty = options.indexOf('checkEmpty') !== -1;
+						optionRequired = options.indexOf('required') !== -1;
+						optionLimit = options.indexOf('limit') !== -1;
+					}
+
+					elements.each(function () {
+						var val = this.value,
+							$this = $(this);
+
+						// check if filed valuer is empty and it's required
+						if (optionRequired && val === '') {
+							$this.addError(requiredError);
+							return true;
```

Comment:
I'll still need to return from function prematurely - I've added `true` just to explicitly say it's not breaking loop.